### PR TITLE
feat(search): Rips out SOLR indexing pipeline

### PR DESCRIPTION
**Pull request by ERosendo**
This PR addresses #4726 

This PR includes the following changes: 

- Removes `scorched` dependency.

- Updates the settings module to clean up SOLR env vars.

- Removes Solr health check

- Removes custom `save` and `as_search_dict` methods from models.

- Refactors AudioFactory class to stop using the `index` argument in hooks.

- Updates admin classes to remove custom methods that were previously used to index data to Solr.

- Removes Solr-specific admin utility functions.

- Removes the normalize_search_dicts helper.

- Renames `solr_list` to `extract_solr_field_values` to better reflect its purpose.

- Removes the following test classes: `EmptySolrTestCase`, `SolrTestCase` and `IndexedSolrTestCase`

- Deletes the following commands:
   - `add_parallel_citations.py`
   - `import_columbia.py`
   - `cl_get_ftm_ids.py`
   - `cl_update_index.py`

### CODE REVIEW

File: cl/citations/tasks.py [hunk 1]
```diff
@@ -214,7 +206,7 @@ def store_opinion_citations_and_update_parentheticals(
                 )
 
     # Finally, commit these changes to the database in a single
-    # transcation block. Trigger a single Solr update as well, if
+    # transcation block. Trigger a single update as well, if
     # required.
```
Comment (mlissner) [inline cl/citations/tasks.py:210]: This whole step is removed, so we can remove the comment about it.

```suggestion
    # transcation block. 
```

File: cl/search/models.py [hunk 2]
```diff
@@ -3075,155 +2891,26 @@ def ordered_opinions(self):
...
-        except NoReverseMatch:
-            raise InvalidDocumentError(
-                "Unable to save to index due to missing absolute_url "
-                "(court_id: %s, item.pk: %s). Might the court have in_use set "
-                "to False?" % (self.docket.court_id, self.pk)
-            )
-
-        # Opinion
-        search_list = []
-        text_template = loader.get_template("indexes/opinion_text.txt")
```
Comment (mlissner) [inline cl/search/models.py:3199]: I think everything in the cl.search.templates.indexes directory can be removed, right?

File: cl/stats/views.py [hunk 3]
```diff
@@ -17,15 +16,13 @@ def health_check(request: HttpRequest) -> JsonResponse:
     """Check if we can connect to various services."""
     is_redis_up = check_redis()
     is_postgresql_up = check_postgresql()
-    is_solr_up = check_solr()
```
Comment (mlissner) [inline cl/stats/views.py:20]: Let's add an Elastic health check here, no?

**Review by mlissner — APPROVED** [review]
Nothing major. A couple more things we can delete, one feature we should upgrade, and I think we're there. Pretty easy to review, really.

File: cl/search/models.py (hunk 2 above)
Comment (ERosendo) [reply cl/search/models.py:3199]: That's correct. I'll remove the directory

File: cl/citations/tasks.py (hunk 1 above)
Comment (ERosendo) [reply cl/citations/tasks.py:210]: alright!

File: cl/stats/views.py (hunk 3 above)
Comment (ERosendo) [reply cl/stats/views.py:20]: Sure! That's a great idea.

**Comment by ERosendo** [comment]
@mlissner I've implemented the changes you suggested.

**Review by mlissner — APPROVED** [review]
AWESOME! 💀

**Pull request merged**